File: server/src/main/java/com/nhom8/services/PublisherService.java
package com.nhom8.services;

import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Sort;
import org.springframework.stereotype.Service;

import com.nhom8.models.Publisher;
import com.nhom8.repository.PublisherRepository;

@Service
public class PublisherService {
    @Autowired
    PublisherRepository publisherRepository;

    public Page<Publisher> getAllPublishers(Optional<Integer> page, Optional<String> sortBy) {
        Page<Publisher> pagePublisher = publisherRepository.findAll(PageRequest.of(page.orElse(0), 5, Sort.Direction.DESC, sortBy.orElse("id")));
        return pagePublisher;
    }

    public Publisher createPublisher(Publisher publisher) {
        return publisherRepository.save(publisher);
    }

    public Publisher getPublisher(Long id) {
        return publisherRepository.findById(id).get();
    }

    public Publisher editPublisher(Long id, Publisher editedPublisher) {
        Publisher publisher = publisherRepository.findById(id).get();
        publisher.setName(editedPublisher.getName());
        publisherRepository.save(publisher);

        return publisher;
    }

    public Publisher deletePublisher(Long id) {
        Publisher publisher = publisherRepository.findById(id).get();            
        publisherRepository.delete(publisher);

        return publisher;
    }
}

File: client/src/services/publisherService.js

import axiosJWT from "../helpers/axiosJWT.js";

const getPublishers = async(currentPage = 0) => {
    const { data } = await axiosJWT.get(`/publishers?page=${currentPage}`);
    return data;
}

const getAllPublishers = async() => {
    const { data } = await axiosJWT.get('/allPublishers');
    return data;
}

const addPublisher = async(body) => {
    return await axiosJWT.post('/publisher', body);
}

const editPublisher = async(id, body) => {
    return await axiosJWT.put(`/publisher/${id}`, body);
}

const deletePublisher = async(id) => {
    return await axiosJWT.delete(`/publisher/${id}`);
}

const publisherService = {
    getPublishers,
    getAllPublishers,
    addPublisher,
    editPublisher,
    deletePublisher
};

export default publisherService;